### PR TITLE
Proposal of rate limit microservice API definition

### PR DESCRIPTION
**Pull request by ralberto**

### CODE REVIEW

**Comment by ralberto** [comment]
Feel free to propose any changes...

**Comment by felipevicens** [comment]
Hello Rui! what do you think if we called it son-gtkrlt instead son-gtkrlimit? to use more or less the same length in the gatekeeper modules names.

File: son-gtkrlimit/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+"check_account_balance_limit" where a client is allowed to perform a maximum of
+10 requests per hour.
+
+```json
+PUT /limits/check_account_balance_limit
+Content-Type: application/json
+{
+  "period": 3600,
+  "limit": 10
+}
```
Comment (jbonnet) [inline son-gtkrlimit/README.md]: A `description` would be helpfull

**Comment by ralberto** [comment]
Hello. Thanks for your comments.
I'm going to change to son-gtkrlt as proposed by @felipevicens  and add an optional field description as proposed by @jbonnet .
 
Yes, the client_id field can be any string. You're free to use an UUID without any problem.
 
> remaining time left to use those remaining number of requests would be nice to
 
I guess it's not clear in the documentation (I'll try to clarify in the docs).  The rate limiter will be implemented using the leaky bucket algorithm. Each client/limit has an empty bucket assigned with a given capacity (according to the limit definition). Each request goes to the bucket, that is leaking requests at a given rate (again according to the limit definition). A client is allowed to perform the request if there are remaining capacity in the bucket. 

Let me give an example:
 
Suppose that we have a limit that allows at maximum 6 requests per minute, that is, period=60 seconds and limit=6 requests. Then our bucket can hold at maximum 6 requests. However we’ll start draining our bucket after “period” at a rate of limit/period, i.e 6req/60sec, that is removing one request from the bucket every 10 seconds. 
 
In our response, the “remaining” field represents the remaining bucket capacity before overflow. The remaining capacity will not expire, so the user can use the remaining capacity anytime.

**Pull request merged**